### PR TITLE
Mutect2 germline resource can have split multiallelic format

### PR DESCRIPTION
**Pull request by davidbenjamin**
See this forum issue: https://gatk.broadinstitute.org/hc/en-us/community/posts/25681424869787-Mutect2-Filtering-Discrepancy-with-Different-Versions-of-gnomAD-Germline-Data

### CODE REVIEW

**Comment by davidbenjamin** [comment]
@jamesemery Not DRAGEN!

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngineUnitTest.java [hunk 1]
```diff
@@ -69,7 +85,7 @@ Object[][] missingAFData() {
 
     @Test(dataProvider = "missingAFData")
     public void testGetGermlineAltAlleleFrequenciesWithMissingAF(final VariantContext vc) {
-        final double[] result = SomaticGenotypingEngine.getGermlineAltAlleleFrequencies(vc.getAlleles(), Optional.of(vc), DEFAULT_AF);
```
Comment (jamesemery) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngineUnitTest.java:72]: Add a test (here or separate) that handles the edge case involving the wrong-length AF list for multiallelics.

**Review by jamesemery — APPROVED** [review]
One test case request that I don't think was covered by the tests you added but otherwise this looks good.

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngineUnitTest.java (hunk 1 above)
Comment (davidbenjamin) [reply src/test/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngineUnitTest.java:72]: done
